Source/Core/DolphinQt/FIFO/FIFOAnalyzer.cpp

```diff
@@ -769,6 +783,227 @@ void FIFOAnalyzer::UpdateDescription()
+      m_gpstate.m_cpmem[i] = file->GetCPMem()[i];
+    for (u32 i = 0; i < FifoDataFile::XF_MEM_SIZE; ++i)
+      m_gpstate.m_xfmem[i] = file->GetXFMem()[i];
+    for (u32 i = 0; i < FifoDataFile::XF_REGS_SIZE; ++i)
+      m_gpstate.m_xfmem[XFMEM_REGISTERS_START + i] = file->GetXFRegs()[i];
+    for (u32 i = 0; i < FifoDataFile::BP_MEM_SIZE; ++i)
+      m_gpstate.m_bpmem[i] = file->GetBPMem()[i];
+  }
+  StateCallback& operator=(StateCallback&& other)
+  {
+    m_gpstate = std::move(other.m_gpstate);
+    m_fifo_pos = std::move(other.m_fifo_pos);
+    return *this;
+  }
+
+  OPCODE_CALLBACK(void OnBP(const u8 command, const u32 value))
+  {
+    u32& mask = m_gpstate.m_bpmem[BPMEM_BP_MASK];
+    u32 oldval = m_gpstate.m_bpmem[command];
+    u32 newval = (oldval & ~mask) | (value & mask);
+    mask = 0xFFFFFF;
+    m_gpstate.m_bpmem[command] = newval;
+    m_fifo_pos.m_bpmem[command] = m_cmd_pos;
+  }
+
+  OPCODE_CALLBACK(void OnCP(const u8 command, const u32 value))
+  {
+    m_cpstate.LoadCPReg(command, value);
+    // LoadCPReg() uses masking in multiple places,
+    // so this is not the same as cpmem[command] = value.
+    m_cpstate.FillCPMemoryArray(m_gpstate.m_cpmem.data());
+    m_fifo_pos.m_cpmem[command] = m_cmd_pos;
+  }
+
+  OPCODE_CALLBACK(void OnXF(const u16 address, const u8 count, const u8* data))
+  {
+    for (u32 i = 0; i < count; ++i)
+    {
+      m_gpstate.m_xfmem[address + i] = Common::swap32(data + sizeof(u32) * i);
+      m_fifo_pos.m_xfmem[address + i] = m_cmd_pos;
+    }
+  }
+
+  OPCODE_CALLBACK(void OnIndexedLoad(const CPArray array, const u32 index, const u16 address,
+                                     const u8 size))
+  {
+    u32 base = m_cpstate.array_bases[array];
+    u32 stride = m_cpstate.array_strides[array];
+    for (u32 i = 0; i < size; ++i)
+    {
+      u32 mem_addr = base + stride * index + sizeof(u32) * i;
+      const auto& mem = mem_addr & 0x1000'0000 ? m_mem2 : m_mem1;
+      m_gpstate.m_xfmem[address + i] = Common::swap32(&mem[mem_addr & ~0x1000'0000]);
+      m_fifo_pos.m_xfmem[address + i] = m_cmd_pos;
+    }
+  }
+
+  OPCODE_CALLBACK(void OnPrimitiveCommand(OpcodeDecoder::Primitive primitive, u8 vat,
+                                          const u32 vertex_size, const u16 num_vertices,
+                                          const u8* vertex_data))
+  {
+  }
+
+  OPCODE_CALLBACK(void OnDisplayList(u32 address, u32 size)) {}
+  OPCODE_CALLBACK(void OnNop(u32 count)) {}
+  OPCODE_CALLBACK(void OnUnknown(u8 opcode, const u8* data)) {}
+  OPCODE_CALLBACK(void OnCommand(const u8* data, u32 size)) {}
+  OPCODE_CALLBACK(CPState& GetCPState()) { return m_cpstate; }
+
+  CPState m_cpstate;
+  GPState m_gpstate;
+
+  // HACK: this stores fifo offsets from the current frame
+  GPState m_fifo_pos{};
+  u32 m_cmd_pos = 0;
+  const std::vector<u8>& m_mem1;
+  const std::vector<u8>& m_mem2;
+};
+
+void FIFOAnalyzer::UpdateStateTree()
+{
+  // FIXME: we sometimes get uninitialized data
```

Comment:
If this is something to be resolved later could you expand on the comment to explain why we're getting uninitialized data?